File: app/src/main/java/com/battaglino/santiago/ubykuo/data/model/ApiResponse.java
package com.battaglino.santiago.ubykuo.data.model;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.util.List;

/**
 * Created by Santiago Battaglino.
 */
public class ApiResponse<T> {

    @SerializedName("total_count")
    @Expose
    public int toalCount;

    @SerializedName("incomplete_results")
    @Expose
    public boolean incompleteResults;

    @SerializedName("items")
    @Expose
    public List<T> items;
}

File: app/src/main/java/com/battaglino/santiago/ubykuo/data/model/ApiResponse.kt
package com.battaglino.santiago.ubykuo.data.model

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created by Santiago Battaglino.
 */
class ApiResponse<T> {

    @SerializedName("total_count")
    @Expose
    var toalCount: Int = 0

    @SerializedName("incomplete_results")
    @Expose
    var incompleteResults: Boolean = false

    @SerializedName("items")
    @Expose
    var items: List<T> = emptyList()
}
